### PR TITLE
Fix(eos_designs): Make it possible to add custom PTP profiles

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Follow up to #4516 

Tackling raising when invalid PTP profiles are referred - this required a couple of changes defined below

## Related Issue(s)

Follow up to  #3930

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

* Remove valid_values for ptp_profiles in various places
* Read default ptp_profiles from schema directly rather than from python file
* raise when an invalid ptp_profile is used in adapters / network_ports / node

- [x] Implement a test where a custom PTP profile is used
- [x] Update release notes

## How to test

molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 1]
```diff
@@ -237,6 +237,17 @@ With AVD version 5.0.0 the default encapsulation used for EVPN iBGP peering betw
 
 See the [porting guide](../porting-guides/5.x.x.md#default-encapsulation-for-wan-evpn-ibgp-peerings-changed-to-path-selection) for details.
 
+#### Allow custom PTP profiles
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: IMO we don't really need to release-note this. It is a fix like many others which are just listed under fixes.

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 2]
```diff
@@ -237,6 +237,17 @@ With AVD version 5.0.0 the default encapsulation used for EVPN iBGP peering betw
...
+The schema was relaxed in various places to allow for custom PTP profiles, before that only the built-in default values were allowed.
+
+- `ptp_settings.profile`
+- `<node_type_key>.defaults.ptp.profile`
+- `<node_type_key>.node_groups[].nodes[].ptp.profile`
+- `<node_type_key>.nodes[].ptp.profile`
+- `<connected_endpoints_key>.adapters[].ptp.profile`
+- `network_ports[].ptp.profile`
+
 #### PREVIEW WAN HA Direct Link now uses Port-Channel by default
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: We should really stop using uppercase preview everywhere.

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/failure-network-port-ptp-profile-does-not-exist.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+network_ports:
+  - switches: [failure-network-port-ptp-profile-does-not-exist]
+    switch_ports: [Ethernet14, Ethernet15]
+    ptp:
+      enabled: true
+      profile: THIS-PROFILE-DOES-NOT-EXIST
+
+expected_error_message: >-
+  Profile 'THIS-PROFILE-DOES-NOT-EXIST' defined under network_ports[0] does not exist in `ptp_profiles`.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/failure-network-port-ptp-profile-does-not-exist.yml]: Not the right place to update it, but here it is easier to see.
```suggestion
  PTP profile 'THIS-PROFILE-DOES-NOT-EXIST' referenced under network_ports[0] does not exist in `ptp_profiles`.
```

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: anyway not for this PR.

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: removed in next commit

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**